x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/use_results_rollup/index.tsx

```diff
@@ -71,49 +75,67 @@ const useStoredPatternRollups = (patterns: string[]) => {
 
     let ignore = false;
     const abortController = new AbortController();
-    const fetchStoredRollups = async () => {
-      const results = await getResults({ httpFetch, abortController, patterns, toasts });
-      if (results?.length && !ignore) {
-        setStoredRollups(Object.fromEntries(results.map(({ rollup }) => [rollup.pattern, rollup])));
+    const fetchStoredPatternResults = async () => {
+      const requests = patterns.map((pattern) =>
+        getStorageResults({ pattern, httpFetch, abortController, toasts }).then((results = []) => ({
+          pattern,
+          results: Object.fromEntries(
+            results.map((storageResult) => [
+              storageResult.indexName,
+              formatResultFromStorage({ storageResult, pattern }),
+            ])
+          ),
+        }))
+      );
+      const patternResults = await Promise.all(requests);
```

Comment:
Right now the API error can not make it to the component, it is caught in the `getStorageResults` on purpose, and that's why we need to pass the `toasts`. I don't think we should add another level of error handling here, it will never catch anything with the current implementation.

> If the implementation of getStorageResults changes to reject when httpFetch() fails, this error will be unhandled by the Promise.all

Agree, but that's a problem only if we change the `getStorageResults` to reject, which is not the case.

Reply:
> Right now the API error can not make it to the component, it is caught in the `getStorageResults` on purpose, and that's why we need to pass the `toasts`. I don't think we should add another level of error handling here, it will never catch anything with the current implementation.
> 
> > If the implementation of getStorageResults changes to reject when httpFetch() fails, this error will be unhandled by the Promise.all
> 
> Agree, but that's a problem only if we change the `getStorageResults` to reject, which is not the case.

Relying on future developers to not change this implementation details of `getStorageResults`, in lieu of handling the error case where `Promise.all` rejects, feels dangerous, because the compiler will not generate an error if a future developer changes the implementation details of `getStorageResults`

I'm wondering, is there a good reason NOT to handle the error case?